File: app/src/main/java/se/gustavkarlsson/skylight/android/evaluation/VisibilityEvaluator.java
package se.gustavkarlsson.skylight.android.evaluation;

import javax.inject.Inject;

import dagger.Reusable;
import se.gustavkarlsson.skylight.android.models.factors.Visibility;

@Reusable
public class VisibilityEvaluator implements ChanceEvaluator<Visibility> {

	@Inject
	VisibilityEvaluator() {
	}

	public Chance evaluate(Visibility visibility) {
		Integer clouds = visibility.getCloudPercentage();
		if (clouds == null) {
			return Chance.unknown();
		}
		return Chance.of((-1.0 / 50.0) * ((double) clouds) + 1.0);
	}
}

File: app/src/main/java/se/gustavkarlsson/skylight/android/evaluation/VisibilityEvaluator.kt
package se.gustavkarlsson.skylight.android.evaluation

import dagger.Reusable
import se.gustavkarlsson.skylight.android.models.factors.Visibility
import javax.inject.Inject

@Reusable
class VisibilityEvaluator
@Inject
internal constructor() : ChanceEvaluator<Visibility> {

    override fun evaluate(value: Visibility): Chance {
        val clouds = value.cloudPercentage ?: return Chance.UNKNOWN
		val chance = -1.0 / 50.0 * clouds.toDouble() + 1.0
		return Chance.of(chance)
    }
}
